### ISSUE TITLE
Implement EnqueueWriteBufferCommand for constant data

### ISSUE THREAD

**Issue by nsmithtt** [tenstorrent/tt-mlir#269]
We want to support embedding constant tensor data in the flatbuffer, the `EnqueueWriteBufferCommand` should be responsible for loading this embedded data onto the device.

Edit/Update with some more thoughts/info:

- You'll have to create a dummy EnqueueWriteBufferCommand op in the flatbuffer + a src tensor that has TensorDesc.constant_data populated.  This field is unused today, but was pre-emptively added for this. 
- Currently nothing goes through EnqueueWriteBufferCommand, we only support tensors coming in through the API, i.e. graph inputs.  Constants differ in this regard because they are baked into the flatbuffer and emitted by the compiler (and opaque to the graph inputs/outputs).
- constant_data is an array of bytes rather which is expected to just be row major packed data. Can be thought of as a void* pointing to the tensor contents
- Should call Metal API `EnqueueWriteBuffer()` with the tensor data.
- We don't have constant tensor support in compiler yet, it's in the works, but there's also missing functionality to actually lower this in the metal dialect too. So for now can just implement/test on runtime side, using `TTMetalToFlatbuffer.cpp` hack to drive a constant tensor locally (don't merge).

**Comment by kmabeeTT** [comment]
@ddilbazTT please take a look at this and start thinking about it as a next task. We can sync offline later today to further understanding.

**Comment by nsmithtt** [comment]
Ok, so after a bit of discussion offline with Kyle, I think we want to change things slightly.
1. Remove `constant_data` from tensor, this doesn't seem like the right place to store this info.
2. Change the `EnqueueWriteBufferCommand` to hold this info instead, something like:
```fbs
table EnqueueWriteBufferCommand {
  broadcast_dims: [int32];
  src: [ubyte];
  dst: TensorRef;
}
```
3. Introduce `broadcast_dims` which will be used to interpret / expand out the `src` field.

Blocked by a minor compiler lowering task (should have review out today):
- https://github.com/tenstorrent/tt-mlir/issues/758

**Comment by kmabeeTT** [comment]
Marking this as closed with PR #791 merged today that adds support for constant scalar FP32 tensors in TTMetal dialect and builds on some slight refactoring from Compiler #758.
